c/tskit/tables.c

```diff
@@ -6758,30 +6790,61 @@ cmp_edge_cl(const void *a, const void *b) {
 /* Squash the edges in the specified array in place. The output edges will
  * be sorted by (child_id, left).
  */
+
 int TSK_WARN_UNUSED
-tsk_squash_edges(tsk_edge_t *edges, size_t num_edges, size_t *num_output_edges)
+tsk_squash_edges(tsk_edge_table_t *edges, size_t num_edges, tsk_size_t *num_output_edges)
 {
     int ret = 0;
     size_t j, k, l;
     tsk_edge_t e;
+    tsk_edge_t edge_list[edges->num_rows];
 
-    qsort(edges, num_edges, sizeof(tsk_edge_t), cmp_edge_cl);
+    // Allocate memory.
+    // tsk_edge_t = calloc(edges->num_rows * sizeof(tsk_edge_t));
+
+    // Turn the edge table into a list of edges.
+    for (k = 0; k < edges->num_rows; k++){
+        e.left = edges->left[k];
+        e.right = edges->right[k];
+        e.parent = edges->parent[k];
+        e.child = edges->child[k];
+        edge_list[k] = e;
+    }
+
+    qsort(edge_list, num_edges, sizeof(tsk_edge_t), cmp_edge_cl);
     j = 0;
     l = 0;
     for (k = 1; k < num_edges; k++) {
-        assert(edges[k - 1].parent == edges[k].parent);
-        if (edges[k - 1].right != edges[k].left || edges[j].child != edges[k].child) {
-            e = edges[j];
-            e.right = edges[k - 1].right;
-            edges[l] = e;
+        assert(edge_list[k-1].parent == edge_list[k].parent);
```

Comment:
I'm not following the logic here --- we can duplicate the logic from the squash_edges function in ``test_topology.c``, right? The only difference here is that we're writing back to the same array rather than creating a second one. This makes it slightly trickier but I'm sure it's possible.